apps/privacy-panel/test/unit/app_list_test.js

```diff
@@ -0,0 +1,166 @@
+            url: 'https://mighty.duck.com/'
+          },
+          permissions: {
+            permission_2: {}
+          }
+        }
+      }
+    ];
+
+    require([
+      'mocks/mock_navigator_moz_apps',
+      'mocks/mock_l10n'
+    ],
+    function(mozApps, mozL10n) {
+      realMozApps = navigator.mozApps;
+      navigator.mozApps = mozApps;
+      navigator.mozApps.mApps = apps;
+
+      realMozL10n = navigator.mozL10n;
+      navigator.mozL10n = mozL10n;
+
+      done();
+    });
+  });
+
+  setup(function(done) {
+    require(['app_list'], appList => {
+      this.subject = appList;
+      done();
+    });
+  });
+
+  suiteTeardown(function() {
+    navigator.mozApps = realMozApps;
+    navigator.mozL10n = realMozL10n;
+  });
+
+  test('initializing, should get all apps', function(done) {
+    this.subject.init().then(function() {
+      var result = this.subject.applications;
+      assert.lengthOf(result, 3);
+      assert.equal(result[0].manifest.name, 'Mozilla Fake App 1');
+      assert.equal(result[1].manifest.name, 'Mozilla Fake App 2');
+      assert.equal(result[2].manifest.name, 'Mozilla Fake App 3');
+      fakeApp1 = result[0];
+      fakeApp2 = result[1];
+      fakeApp3 = result[2];
+      done();
+    }.bind(this));
+  });
+
+  test('should get one app', function(done) {
+    var result = this.subject.getFilteredApps('permission_1');
+    assert.lengthOf(result, 1);
+    assert.equal(result[0].manifest.name, 'Mozilla Fake App 1');
+    done();
+  });
+
+  test('should get a list of apps', function(done) {
+    var result = this.subject.getFilteredApps('permission_2');
+    assert.lengthOf(result, 2);
+    assert.equal(result[0].manifest.name, 'Mozilla Fake App 2');
+    assert.equal(result[1].manifest.name, 'Mozilla Fake App 3');
+    done();
+  });
+
+  test('should get an empty list of apps for an invalid permission',
+    function(done) {
+      var result = this.subject.getFilteredApps('invalidPermission');
+      assert.isArray(result);
+      assert.lengthOf(result, 0);
+      done();
+    }
+  );
+
+  test('should get path to app icon',
+    function(done) {
+      assert.notEqual(fakeApp1.iconURL, '../style/images/default.png');
+      done();
+    }
+  );
+
+  test('should get default path to app icon if path is not set in manifest',
+    function(done) {
+      assert.equal(fakeApp2.iconURL, '../style/images/default.png');
+      done();
+    }
+  );
+
+  test('should get an array of app groups (one group per trust level)',
+    function(done) {
+      assert.deepEqual(this.subject.getSortedApps('trust'), {
+        'privileged': [ fakeApp1, fakeApp2 ],
+        'web': [ fakeApp3 ]
+      });
+      done();
+    }
+  );
+
+  test('should get an array of app groups (one group per vendor)',
+    function(done) {
+      assert.deepEqual(this.subject.getSortedApps('vendor'), {
+        'Flying Platypus': [ fakeApp1 ],
+        'Mighty Duck': [ fakeApp2, fakeApp3 ]
+      });
+      done();
+    }
+  );
```

Comment:
The test just above this one sorts apps by `app.trust`, just like this test sorts apps by `app.vendor`. Please clarify if you think these tests aren’t meaningful.